### ISSUE TITLE
Sometimes training doesn't converge 

### ISSUE THREAD

**Issue by nwang2** [intel-analytics/BigDL#1486]
We use latest git head 6602a8f6e8d7c550b363501e83fc20a28b8f6342 to do inception V1 and Resnet training, and find that training doesn't converge randomly:
# grep Loss bigdl.log | tail -n 10
2017-08-21 07:07:11 INFO  DistriOptimizer$:295 - [Epoch 82 184576/1281167][Iteration 58019][Wall Clock 251117.710963265s] Train 1792 in 4.220951966seconds. Throughput is 424.54877 records/second. Loss is NaN. Current learning rate is 0.022707158569988024. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:15 INFO  DistriOptimizer$:295 - [Epoch 82 186368/1281167][Iteration 58020][Wall Clock 251121.931915231s] Train 1792 in 4.147252403seconds. Throughput is 432.0933 records/second. Loss is NaN. Current learning rate is 0.022704307165623168. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:19 INFO  DistriOptimizer$:295 - [Epoch 82 188160/1281167][Iteration 58021][Wall Clock 251126.079167634s] Train 1792 in 4.264490471seconds. Throughput is 420.21432 records/second. Loss is NaN. Current learning rate is 0.022701455403109177. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:23 INFO  DistriOptimizer$:295 - [Epoch 82 189952/1281167][Iteration 58022][Wall Clock 251130.343658105s] Train 1792 in 4.155292104seconds. Throughput is 431.2573 records/second. Loss is NaN. Current learning rate is 0.02269860328231104. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:28 INFO  DistriOptimizer$:295 - [Epoch 82 191744/1281167][Iteration 58023][Wall Clock 251134.498950209s] Train 1792 in 4.492550308seconds. Throughput is 398.88257 records/second. Loss is NaN. Current learning rate is 0.022695750803093717. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:32 INFO  DistriOptimizer$:295 - [Epoch 82 193536/1281167][Iteration 58024][Wall Clock 251138.991500517s] Train 1792 in 4.197963127seconds. Throughput is 426.8737 records/second. Loss is NaN. Current learning rate is 0.022692897965322013. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:36 INFO  DistriOptimizer$:295 - [Epoch 82 195328/1281167][Iteration 58025][Wall Clock 251143.189463644s] Train 1792 in 4.15345278seconds. Throughput is 431.44824 records/second. Loss is NaN. Current learning rate is 0.022690044768860714. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:40 INFO  DistriOptimizer$:295 - [Epoch 82 197120/1281167][Iteration 58026][Wall Clock 251147.342916424s] Train 1792 in 4.074771818seconds. Throughput is 439.7792 records/second. Loss is NaN. Current learning rate is 0.02268719121357446. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:44 INFO  DistriOptimizer$:295 - [Epoch 82 198912/1281167][Iteration 58027][Wall Clock 251151.417688242s] Train 1792 in 3.758637876seconds. Throughput is 476.76846 records/second. Loss is NaN. Current learning rate is 0.02268433729932787. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-21 07:07:48 INFO  DistriOptimizer$:295 - [Epoch 82 200704/1281167][Iteration 58028][Wall Clock 251155.176326118s] Train 1792 in 4.106297326seconds. Throughput is 436.40286 records/second. Loss is NaN. Current learning rate is 0.02268148302598541. Current weight decay is 1.0E-4. Current momentum is 0.9.

**Comment by nwang2** [comment]
After going through the whole log, we see the training did converge until iteration 45459 where the loss became infinity:
2017-08-20 16:11:38 INFO  DistriOptimizer$:295 - [Epoch 64 736512/1281167][Iteration 45457][Wall Clock 197331.83477254s] Train 1792 in 4.387814632seconds. Throughput is 408.40378 records/second. Loss is 1.2645824. Current learning rate is 0.04628416419774555. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:11:42 INFO  DistriOptimizer$:295 - [Epoch 64 738304/1281167][Iteration 45458][Wall Clock 197336.222587172s] Train 1792 in 4.494659207seconds. Throughput is 398.6954 records/second. Loss is 1.308916. Current learning rate is 0.046282765356364115. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:11:46 INFO  DistriOptimizer$:295 - [Epoch 64 740096/1281167][Iteration 45459][Wall Clock 197340.717246379s] Train 1792 in 4.006890034seconds. Throughput is 447.22968 records/second. Loss is Infinity. Current learning rate is 0.046281366472703075. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:11:51 INFO  DistriOptimizer$:295 - [Epoch 64 741888/1281167][Iteration 45460][Wall Clock 197344.724136413s] Train 1792 in 4.253184791seconds. Throughput is 421.33133 records/second. Loss is 1.2483193. Current learning rate is 0.04627996754675863. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:11:55 INFO  DistriOptimizer$:295 - [Epoch 64 743680/1281167][Iteration 45461][Wall Clock 197348.977321204s] Train 1792 in 4.253003626seconds. Throughput is 421.34927 records/second. Loss is 1.2952143. Current learning rate is 0.04627856857852692. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:11:59 INFO  DistriOptimizer$:295 - [Epoch 64 745472/1281167][Iteration 45462][Wall Clock 197353.23032483s] Train 1792 in 4.23005562seconds. Throughput is 423.63507 records/second. Loss is 1.3166977. Current learning rate is 0.04627716956800412. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:12:04 INFO  DistriOptimizer$:295 - [Epoch 64 747264/1281167][Iteration 45463][Wall Clock 197357.46038045s] Train 1792 in 4.568494111seconds. Throughput is 392.25177 records/second. Loss is 1.2860057. Current learning rate is 0.04627577051518638. Current weight decay is 1.0E-4. Current momentum is 0.9.
.....
2017-08-20 16:13:55 INFO  DistriOptimizer$:295 - [Epoch 64 793856/1281167][Iteration 45489][Wall Clock 197468.891525812s] Train 1792 in 4.344207943seconds. Throughput is 412.5033 records/second. Loss is 1.2230562. Current learning rate is 0.04623938028382581. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:13:59 INFO  DistriOptimizer$:295 - [Epoch 64 795648/1281167][Iteration 45490][Wall Clock 197473.235733755s] Train 1792 in 4.312448926seconds. Throughput is 415.54114 records/second. Loss is Infinity. Current learning rate is 0.046237980087592986. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:04 INFO  DistriOptimizer$:295 - [Epoch 64 797440/1281167][Iteration 45491][Wall Clock 197477.548182681s] Train 1792 in 4.518316646seconds. Throughput is 396.60788 records/second. Loss is 1.2653805. Current learning rate is 0.0462365798489576. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:08 INFO  DistriOptimizer$:295 - [Epoch 64 799232/1281167][Iteration 45492][Wall Clock 197482.066499327s] Train 1792 in 4.285363497seconds. Throughput is 418.16754 records/second. Loss is 1.3053167. Current learning rate is 0.046235179567915806. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:12 INFO  DistriOptimizer$:295 - [Epoch 64 801024/1281167][Iteration 45493][Wall Clock 197486.351862824s] Train 1792 in 4.276199933seconds. Throughput is 419.06366 records/second. Loss is Infinity. Current learning rate is 0.046233779244463744. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:17 INFO  DistriOptimizer$:295 - [Epoch 64 802816/1281167][Iteration 45494][Wall Clock 197490.628062757s] Train 1792 in 4.236683391seconds. Throughput is 422.97238 records/second. Loss is 1.3417698. Current learning rate is 0.04623237887859755. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:21 INFO  DistriOptimizer$:295 - [Epoch 64 804608/1281167][Iteration 45495][Wall Clock 197494.864746148s] Train 1792 in 4.399126511seconds. Throughput is 407.3536 records/second. Loss is Infinity. Current learning rate is 0.046230978470313384. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-20 16:14:25 INFO  DistriOptimizer$:295 - [Epoch 64 806400/1281167][Iteration 45496][Wall Clock 197499.263872659s] Train 1792 in 4.083894503seconds. Throughput is 438.7968 records/second. Loss is 1.31938. Current learning rate is 0.046229578019607406. Current weight decay is 1.0E-4. Current momentum is 0.9.

**Comment by nwang2** [comment]
We tried to resume the training from the checkpoint before the "loss infinity" by adding --state optimMethod.45260 --model model.45260 to the command line. But the issue still reproduced:
2017-08-22 00:39:52 INFO  DistriOptimizer$:295 - [Epoch 76 996352/1281167][Iteration 54396][Wall Clock 39929.188837978s] Train 1792 in 4.457925353seconds. Throughput is 401.9807 records/second. Loss is 1.1553274. Current learning rate is 0.03138063837011811. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 00:39:56 INFO  DistriOptimizer$:295 - [Epoch 76 998144/1281167][Iteration 54397][Wall Clock 39933.646763331s] Train 1792 in 4.087660204seconds. Throughput is 438.39258 records/second. Loss is 1.1260583. Current learning rate is 0.03137857514394999. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 00:40:00 INFO  DistriOptimizer$:295 - [Epoch 76 999936/1281167][Iteration 54398][Wall Clock 39937.734423535s] Train 1792 in 4.30715078seconds. Throughput is 416.0523 records/second. Loss is 1.2019689. Current learning rate is 0.03137651178211025. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 00:40:04 INFO  DistriOptimizer$:295 - [Epoch 76 1001728/1281167][Iteration 54399][Wall Clock 39942.041574315s] Train 1792 in 4.232361321seconds. Throughput is 423.4043 records/second. Loss is Infinity. Current learning rate is 0.031374448284572115. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 00:40:09 INFO  DistriOptimizer$:295 - [Epoch 76 1003520/1281167][Iteration 54400][Wall Clock 39946.273935636s] Train 1792 in 4.458929152seconds. Throughput is 401.89023 records/second. Loss is 1.0660006. Current learning rate is 0.03137238465130883. Current weight decay is 1.0E-4. Current momentum is 0.9.
......
2017-08-22 04:15:08 INFO  DistriOptimizer$:295 - [Epoch 80 1184512/1281167][Iteration 57361][Wall Clock 52847.236253558s] Train 1792 in 3.93863422seconds. Throughput is 454.98007 records/second. Loss is Infinity. Current learning rate is 0.024511571460921696. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:13 INFO  DistriOptimizer$:295 - [Epoch 80 1186304/1281167][Iteration 57362][Wall Clock 52851.174887778s] Train 1792 in 4.218363287seconds. Throughput is 424.8093 records/second. Loss is NaN. Current learning rate is 0.024508929985461593. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:17 INFO  DistriOptimizer$:295 - [Epoch 80 1188096/1281167][Iteration 57363][Wall Clock 52855.393251065s] Train 1792 in 4.034650545seconds. Throughput is 444.1525 records/second. Loss is NaN. Current learning rate is 0.02450628822528302. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:21 INFO  DistriOptimizer$:295 - [Epoch 80 1189888/1281167][Iteration 57364][Wall Clock 52859.42790161s] Train 1792 in 4.18154777seconds. Throughput is 428.54947 records/second. Loss is NaN. Current learning rate is 0.0245036461802939. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:25 INFO  DistriOptimizer$:295 - [Epoch 80 1191680/1281167][Iteration 57365][Wall Clock 52863.60944938s] Train 1792 in 4.153980561seconds. Throughput is 431.39343 records/second. Loss is NaN. Current learning rate is 0.024501003850402114. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:29 INFO  DistriOptimizer$:295 - [Epoch 80 1193472/1281167][Iteration 57366][Wall Clock 52867.763429941s] Train 1792 in 4.011746594seconds. Throughput is 446.68826 records/second. Loss is NaN. Current learning rate is 0.024498361235515447. Current weight decay is 1.0E-4. Current momentum is 0.9.
2017-08-22 04:15:33 INFO  DistriOptimizer$:295 - [Epoch 80 1195264/1281167][Iteration 57367][Wall Clock 52871.775176535s] Train 1792 in 4.131840622seconds. Throughput is 433.70502 records/second. Loss is NaN. Current learning rate is 0.024495718335541698. Current weight decay is 1.0E-4. Current momentum is 0.9.

**Comment by qiuxin2012** [comment]
As our testing, Infinity means a loss bigger than 1e7. 
Could you also attach the running `spark-submit` command? @nwang2

**Comment by nwang2** [comment]
spark-submit --class com.intel.analytics.bigdl.models.inception.TrainInceptionV1 --master spark://hsx-node1:7077 --driver-memory=30g --executor-cores=28 --num-executors=8 --total-executor-cores=224 --executor-memory=220g --driver-java-options="-Dlog4j.configuration=/mnt/disk5/BigDL/log4j.conf" /opt/package/BigDL/dist/lib/bigdl-0.3.0-SNAPSHOT-jar-with-dependencies.jar --batchSize 1792 -f hdfs:///user/root/sequence/ --checkpoint /mnt/disk2/inceptionV1_Spark2.1.1_standalone/ --learningRate 0.0896
